### PR TITLE
Convert Markdown frontmatter to YAML

### PR DESCRIPTION
**Pull request by mstanleyjones**
Some frontmatter such as the weights, menu stuff, etc is no longer used
'draft=true' becomes 'published: false'

Signed-off-by: Misty Stanley-Jones <misty@docker.com>

### CODE REVIEW

**Comment by mstanleyjones** [comment]
@thaJeztah PTAL

File: docs/spec/api.md [hunk 1]
```diff
@@ -1,12 +1,8 @@
...
-keywords = ["registry, on-prem, images, tags, repository, distribution, api, advanced"]
-[menu.main]
-parent="smn_registry_ref"
-+++
-<![end-metadata]-->
+---
+title: "HTTP API V2"
+description: "Specification for the Registry API."
+keywords: ["registry, on-prem, images, tags, repository, distribution, api, advanced"]
+---
```
Comment (thaJeztah) [inline docs/spec/api.md]: This probably needs to be updated here as well; https://github.com/docker/distribution/blob/master/docs/spec/api.md.tmpl#L1-L9

(or instead: @stevvooe ?)

**Comment by mstanleyjones** [comment]
Good catch, I missed that one. I'll update it too.

File: docs/spec/api.md (hunk 1 above)
Comment (RichardScothern) [inline docs/spec/api.md]: It should be changed in both places otherwise, the change will be lost the next time `api.md` is generated.

**Comment by mstanleyjones** [comment]
Should be good now.

**Comment by mstanleyjones** [comment]
It is, I'm pretty sure.

**Comment by RichardScothern** [comment]
Build breakage unrelated to this change.  LGTM.

**Pull request merged**